### PR TITLE
refactor(migrations): packageRules

### PR DESCRIPTION
**Pull request by pret-a-porter**
## Changes:

One more migration from previously huge PR https://github.com/renovatebot/renovate/pull/12957

## Context:

1. https://github.com/renovatebot/renovate/issues/11459
2. Part of https://github.com/renovatebot/renovate/pull/12957 

## Documentation

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/migration.ts [hunk 1]
```diff
@@ -68,11 +68,6 @@ export function migrateConfig(
           migratedConfig[newKey] = true;
         }
         delete migratedConfig[key];
-      } else if (key === 'packageRules' && is.plainObject(val)) {
```
Comment (viceice) [inline lib/config/migration.ts:71]: 🧐why do we have such a migration? 🤔🙈

File: lib/config/migrations/custom/package-rules-migration.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import is from '@sindresorhus/is';
+import { AbstractMigration } from '../base/abstract-migration';
+
+export class PackageRulesMigration extends AbstractMigration {
+  override readonly propertyName = 'packageRules';
+
+  override run(value: unknown): void {
+    const packageRules = this.get('packageRules');
+
+    if (is.plainObject(value)) {
```
Comment (viceice) [inline lib/config/migrations/custom/package-rules-migration.ts:10]: Please add a test for else condition for better branch coverage.

File: lib/config/migration.ts (hunk 1 above)
Comment (pret-a-porter) [reply lib/config/migration.ts:71]: No idea, maybe some of migrations already can be removed

Comment (rarkins) [reply lib/config/migration.ts:71]: I think I saw some people with a single packageRule mistakenly configured like this:

```json
{
  "packageRules": {
    "matchDatasources": ["maven"],
    "labels": ["maven"]
  }
}
```

We *could* drop this migration and give them an error instead.

Comment (pret-a-porter) [reply lib/config/migration.ts:71]: Migration removed

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
Coverage problem

**Review by rarkins — APPROVED** [review]

**Pull request merged**